### PR TITLE
Roll analyzer, fix package loading, run side by side

### PR DESCRIPTION
**Pull request by samuelli**
Part of #913 

Summary of key changes:
 * Rolls polymer-analyzer to latest
 * Fixes package loading to use a root path to where it was installed and using relative paths for `mainHtmlPaths`.
 * Package analysis looks through main HTML paths & expands out imported documents while staying within the package.
 * Duplicates `<catalog-element>`, which switches loading to an analyzer supporting version based on the query param `?analyzer`. This means once its ready, we'll switch over directly to the new one without any code debt.

### CODE REVIEW

**Comment by samuelli** [comment]
@andymutton: can you look at `analysis.js`? Particularly where I assume that components are installed under `attributes.repo` in the current directory of the process.
@dstockwell: can you look at the rest of the analysis changes? The client/ change is with the duplication & rewrites of `_elements()`, `_behaviors()`. The sidebar populates with elements & behaviors, but otherwise, its not really functioning correctly, but it'll be easier to review once we land the duplicated file, so you can actually see diffs.

File: client/src/catalog-app.html [hunk 1]
```diff
@@ -690,6 +701,7 @@ <h2 id="domain-title">webcomponents.org</h2>
 
         if (isLocalhost)
           result.api = 'https://www.webcomponents.org';
+        result.api = 'https://custom-elements-staging.appspot.com';
```
Comment (dstockwell) [inline client/src/catalog-app.html]: is this meant to be here?

File: analysis/src/analysis.js [hunk 2]
```diff
@@ -46,9 +47,11 @@ class Analysis {
       this.bower.prune().then(() => {
         return this.bower.install(attributes.owner, attributes.repo, versionOrSha);
       }).then(mainHtmlPaths => {
+        const root = path.resolve(process.cwd(), 'bower_components', attributes.repo);
```
Comment (andymutton) [inline analysis/src/analysis.js]: Yeah... I wish I could be more helpful here, but I'm not 100% sure that the attributes.repo assumption is correct and I'm also not 100% sure that it's not. My memory on this is fuzzy at best.

Is there a way to verify the correctness with an assert here and then just run a full analysis on staging?

File: analysis/src/analyzer.js [hunk 3]
```diff
@@ -28,9 +28,14 @@ class AnalyzerRunner {
 
       if (inputs == null || inputs.length === 0) {
         resolve({});
+        // TODO: fall back to package analysis
       } else {
         Promise.all(inputs.map((i) => analyzer.analyze(i))).then(function(documents) {
-          resolve(generateAnalysis(documents, ''));
+          var allDocuments = [];
+          documents.forEach(function(doc) {
+            allDocuments = allDocuments.concat(Array.from(doc.getByKind('document', {imported: true, externalPackages: false})));
```
Comment (dstockwell) [inline analysis/src/analyzer.js]: so this still includes all of the main documents?

Comment (samuelli) [inline analysis/src/analyzer.js]: Yes.

Based on offline convos, we were going to ignore main documents & search the entire package, but this turned out to include the wrong things for packages like `app-layout`, which include all sorts of elements for testing, demos, sites etc.

This instead analyses all the main documents, and only includes everything they import while still in the same package.

File: client/src/catalog-app.html (hunk 1 above)
Comment (samuelli) [inline client/src/catalog-app.html]: No. Thanks.

File: analysis/src/analysis.js (hunk 2 above)
Comment (samuelli) [inline analysis/src/analysis.js]: Sure, I'll do that.

**Pull request merged**